vscode/core/src/utilities/tls.ts

```diff
@@ -1,24 +1,47 @@
 import tls from "node:tls";
 import fs from "fs/promises";
-import { Agent as HttpsAgent } from "node:https";
-import { Agent as UndiciAgent } from "undici";
+import { Agent as HttpsAgent, type AgentOptions } from "node:https";
+import { Agent as UndiciAgent, ProxyAgent } from "undici";
 import type { Dispatcher as UndiciTypesDispatcher } from "undici-types";
+import { NodeHttpHandler, NodeHttp2Handler } from "@smithy/node-http-handler";
+import { HttpsProxyAgent } from "https-proxy-agent";
+import type { Logger } from "winston";
 
 export async function getDispatcherWithCertBundle(
   bundlePath: string | undefined,
   insecure: boolean = false,
+  allowH2: boolean = false,
 ): Promise<UndiciTypesDispatcher> {
   let allCerts: string | undefined;
   if (bundlePath) {
     const defaultCerts = tls.rootCertificates.join("\n");
     const certs = await fs.readFile(bundlePath, "utf8");
     allCerts = [defaultCerts, certs].join("\n");
   }
+
+  const proxyUrl =
+    process.env.HTTPS_PROXY ||
+    process.env.https_proxy ||
+    process.env.HTTP_PROXY ||
+    process.env.http_proxy;
+
+  if (proxyUrl) {
+    return new ProxyAgent({
+      uri: proxyUrl,
+      allowH2,
+      connect: {
+        ca: allCerts,
+        rejectUnauthorized: !insecure,
+      },
+    }) as unknown as UndiciTypesDispatcher;
+  }
+
   return new UndiciAgent({
     connect: {
       ca: allCerts,
       rejectUnauthorized: !insecure,
     },
+    allowH2,
```

Comment:
By default (http1 setting), every request ("fetch based providers" ) goes through the custom dispatcher with allowH2: false, which forces HTTP/1.1